propagator/propagator.go

```diff
@@ -56,11 +62,9 @@ func (p CloudTraceFormatPropagator) Inject(ctx context.Context, carrier propagat
 	// https://cloud.google.com/trace/docs/setup#force-trace
 	// Trace ID: 32-char hexadecimal value representing a 128-bit number.
 	// Span ID: decimal representation of the unsigned interger.
-	sidHex := sc.SpanID().String()
-	sid, err := strconv.ParseUint(sidHex, 16, 64)
-	if err != nil {
-		return
-	}
+	ary := sc.SpanID()
+	sid := binary.BigEndian.Uint64(ary[:])
```

Comment:
Is the endian-ness a publicly documented part of the SpanID API?

Reply:
I was also not sure on this hoping for other's inputs, but my guess is "not applicable". OTel spec says:

> [`SpanId` A valid span identifier is an 8-byte array with at least one non-zero byte.](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#:~:text=SpanId%20A%20valid%20span%20identifier%20is%20an%208-byte%20array%20with%20at%20least%20one%20non-zero%20byte.)

Since it's just defined as a byte array, there is no endianness or MSB/LSB right? The XCTC context spec just says

> [`SPAN_ID` is the decimal representation of the (unsigned) span ID.](https://cloud.google.com/trace/docs/setup#:~:text=SPAN_ID%20is%20the%20decimal%20representation%20of%20the%20(unsigned)%20span%20ID.)

My take is it doesn't matter as long as we are consistent across implementations on how we interpret the byte array as uint64. Wdyt?

Reply:
I agree, it doesn't matter as long as we're consistent across implementations. I'm happy to approve and merge this PR, but in the near future we should create a doc about XCTC-OTel requirements so that we can actually keep this consistent.